data-plane/dispatcher/src/test/java/dev/knative/eventing/kafka/broker/dispatcher/ConsumerDeployerVerticleTest.java

```diff
@@ -44,33 +45,38 @@
 
 @ExtendWith(VertxExtension.class)
 @Execution(ExecutionMode.CONCURRENT)
-public class ConsumerDeployerTest {
+public class ConsumerDeployerVerticleTest {
 
   @Test
   @Timeout(value = 2)
-  public void shouldAddResourceAndDeployVerticles(final Vertx vertx, final VertxTestContext context) {
+  public void shouldAddResourceAndDeployVerticles(final Vertx vertx, final VertxTestContext context)
+    throws ExecutionException, InterruptedException {
     final var resources = List.of(
       resource1(),
       resource2()
     );
     final var numEgresses = numEgresses(resources);
     final var checkpoints = context.checkpoint(1);
 
-    final var consumerDeployer = new ConsumerDeployer(
-      vertx,
+    final var consumerDeployer = new ConsumerDeployerVerticle(
       (resource, egress) -> new AbstractVerticle() {
       },
       100
     );
 
+    vertx.deployVerticle(consumerDeployer)
+      .toCompletionStage()
+      .toCompletableFuture()
+      .get();
+
     final var reconciler = ResourcesReconcilerImpl
       .builder()
       .watchEgress(consumerDeployer)
       .build();
 
     reconciler.reconcile(resources)
       .onSuccess(ignored -> context.verify(() -> {
-        assertThat(vertx.deploymentIDs()).hasSize(numEgresses);
+        assertThat(vertx.deploymentIDs()).hasSize(numEgresses + 1);
```

Comment:
Add a comment for the `+1` or change it `+ NUM_SYSTEM_VERTICLES` or something similar.